src/transformers/models/sam_hq/modular_sam_hq.py

```diff
@@ -0,0 +1,1170 @@
+    def __init__(self, config: SamHQMaskDecoderConfig):
+        super().__init__()
+        self.hidden_size = config.hidden_size
+
+        self.num_multimask_outputs = config.num_multimask_outputs
+        self.num_mask_tokens = config.num_multimask_outputs + 1
+
+        self.iou_token = nn.Embedding(1, self.hidden_size)
+        self.mask_tokens = nn.Embedding(self.num_mask_tokens, self.hidden_size)
+
+
+        self.transformer = SamHQTwoWayTransformer(config)
+
+        self.upscale_conv1 = nn.ConvTranspose2d(self.hidden_size, self.hidden_size // 4, kernel_size=2, stride=2)
+        self.upscale_conv2 = nn.ConvTranspose2d(self.hidden_size // 4, self.hidden_size // 8, kernel_size=2, stride=2)
+        self.upscale_layer_norm = SamHQLayerNorm(self.hidden_size // 4, data_format="channels_first")
+        self.activation = nn.GELU()
+
+
+        mlps_list = []
+        for _ in range(self.num_mask_tokens):
+            mlps_list += [SamHQFeedForward(self.hidden_size, self.hidden_size, self.hidden_size // 8, 3)]
+        self.output_hypernetworks_mlps = nn.ModuleList(mlps_list)
+
+        self.iou_prediction_head = SamHQFeedForward(
+            self.hidden_size, config.iou_head_hidden_dim, self.num_mask_tokens, config.iou_head_depth
+        )
+
+
+        self.hq_token = nn.Embedding(1, self.hidden_size)
+        self.hf_mlp = SamHQFeedForward(self.hidden_size, self.hidden_size, self.hidden_size // 8, 3)
+        self.num_mask_tokens = self.num_mask_tokens + 1
+
+        # Compress ViT features
+        self.compress_vit_conv1 = nn.ConvTranspose2d(config.vit_dim, self.hidden_size, kernel_size=2, stride=2)
+        self.compress_vit_norm = SamHQLayerNorm(self.hidden_size, data_format="channels_first")
+        self.compress_vit_conv2 = nn.ConvTranspose2d(self.hidden_size, self.hidden_size // 8, kernel_size=2, stride=2)
+        self.activation = nn.GELU()
+
+        # Embedding encoder
+        self.encoder_conv1 = nn.ConvTranspose2d(self.hidden_size, self.hidden_size // 4, kernel_size=2, stride=2)
+        self.encoder_norm = SamHQLayerNorm(self.hidden_size // 4, data_format="channels_first")
+        self.encoder_conv2 = nn.ConvTranspose2d(self.hidden_size // 4, self.hidden_size // 8, kernel_size=2, stride=2)
+
+        # Embedding mask feature
+        self.mask_conv1 = nn.Conv2d(self.hidden_size // 8, self.hidden_size // 4, kernel_size=3, stride=1, padding=1)
+        self.mask_norm = SamHQLayerNorm(self.hidden_size // 4, data_format="channels_first")
+        self.mask_conv2 = nn.Conv2d(self.hidden_size // 4, self.hidden_size // 8, kernel_size=3, stride=1, padding=1)
+
+
+    def forward(
+            self,
+            image_embeddings: torch.Tensor,
+            image_positional_embeddings: torch.Tensor,
+            sparse_prompt_embeddings: torch.Tensor,
+            dense_prompt_embeddings: torch.Tensor,
+            multimask_output: bool,
+            hq_token_only: bool,
+            interm_embeddings: torch.Tensor,
```

Comment:
in general, let's avoid abbreviations
```suggestion
            intermediate_embeddings: torch.Tensor,
```

Reply:
done